stl/inc/xtree

```diff
@@ -1625,39 +1622,36 @@ protected:
         }
     }
 
-    template <class _Ty, class _Is_set>
-    _Nodeptr _Copy_or_move(_Ty& _Val, _Copy_tag, _Is_set) { // copy to new node
-        return _Buynode(_Val);
-    }
-
-    template <class _Ty>
-    _Nodeptr _Copy_or_move(_Ty& _Val, _Move_tag, true_type) { // move to new node -- set
-        return _Buynode(_STD move(_Val));
-    }
-
-    template <class _Ty>
-    _Nodeptr _Copy_or_move(_Ty& _Val, _Move_tag, false_type) { // move to new node -- map
-        return _Buynode(_STD move(const_cast<key_type&>(_Val.first)), _STD move(_Val.second));
+    template <_Copy_or_move_t _Copy_or_move_v, class _Ty>
+    _Nodeptr _Copy_or_move(_Ty& _Val) {
+        if constexpr (_Copy_or_move_v == _Copy_or_move_t::_Copy) {
+            return _Buynode(_Val);
+        } else {
+            if constexpr (_Is_set) {
+                return _Buynode(_STD move(_Val));
+            } else {
+                return _Buynode(_STD move(const_cast<key_type&>(_Val.first)), _STD move(_Val.second));
+            }
+        }

```

Comment:
I think you can flatten that into one else if
```suggestion
        } else if constexpr (_Is_set) {
            return _Buynode(_STD move(_Val));
        } else {
            return _Buynode(_STD move(const_cast<key_type&>(_Val.first)), _STD move(_Val.second));
        }
```

Reply:
I think unflatten is more clear

Reply:
They're pretty close, but I agree with Alex that the unflattened structure is clearer. The rationale is that this is a forked path, with a sub-fork, which is shown by the unflattened structure. It would be equally clear (ignoring our "prefer positive tests" convention) if the branches were reversed. Flattening the logic would make the order of the tests important, and having to worry about the order in which tests are performed is a slight mental burden.

Reply:
I also have a mild preference to unflatten, so it's a tie, and the status quo code in the PR stays.